File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/Proxy/IProxyListProvider.java
package ru.kmorozov.gbd.core.logic.Proxy;

import java.util.stream.Stream;

/**
 * Created by km on 27.11.2015.
 */
public interface IProxyListProvider {

    void invalidatedProxyListener();

    void updateProxyList();

    Stream<HttpHostExt> getParallelProxyStream();

    int getProxyCount();

    void processProxyList(UrlType urlType);

    Iterable<HttpHostExt> getProxyList();

    boolean proxyListCompleted();
}

File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/proxy/IProxyListProvider.kt
package ru.kmorozov.gbd.core.logic.proxy

import java.util.stream.Stream

/**
 * Created by km on 27.11.2015.
 */
interface IProxyListProvider {

    val parallelProxyStream: Stream<HttpHostExt>

    val proxyCount: Int

    val proxyList: Iterable<HttpHostExt>

    fun invalidatedProxyListener()

    fun updateProxyList()

    fun processProxyList(urlType: UrlType)
}
